src/races/races.controller.ts

```diff
@@ -0,0 +1,60 @@
+import { Controller, Get, Query, NotFoundException } from '@nestjs/common'
+import { RacesService } from './races.service'
+import { NormalizedRace } from './races.types'
+import { ZodValidationPipe } from 'nestjs-zod'
+import { RacesListQuery, racesListSchema } from './schemas/racesList.schema'
+
+@Controller('races')
+export class RacesController {
+  constructor(private readonly racesService: RacesService) {}
+
+  @Get()
+  async findRaces(
+    @Query(new ZodValidationPipe(racesListSchema)) query: RacesListQuery,
+  ): Promise<NormalizedRace | NormalizedRace[]> {
+    const { state, county, city, positionSlug } = query
+    if (state && county && city && positionSlug) {
+      const race = await this.racesService.findOne(
+        state,
+        county,
+        city,
+        positionSlug,
+      )
+      if (!race) {
+        throw new NotFoundException('Race not found')
+      }
+      return race
```

Comment:
Also, I noticed this later.. We really should keep the type interface on any given endpoint consistent.

That means, if an endpoint returns an `[]` of resources for a result that contains many resources, then it should _always_ return that data structure. Even if there's only 1 resource instance in the results, it's still, `[resource]`, and in the event that the filters provided result in zero results, the return ends up being 200 OK, `[]`, empty array, not 404.

404 is an HTTP status. It's saying to the client, essentially, "the address of the resource you're looking for doesn't exist", which would be the case for an endpoint like, `/races/:raceId`.  If there is no race resource that exists for the given `raceId`, then that URI, or URL, just doesn't plain exist on the internet, so that results in a 404.

But in the case of `/races?[queryFilters]`, even if the filters provided results in an `[]`, then that doesn't negate the fact that `/races` is still a valid path to a Universal Resource Identifier/Location, so that would still be a 200, and not a 404.